### DISCUSSION TITLE
Could the JDBC wrapper work as a standalone JDBC driver for Cassandra?

### DISCUSSION THREAD

**Discussion by fiddleboy** [ing-bank/cassandra-jdbc-wrapper#4] [Q&A]
Hello, i'm kind of a newbie in JDBC driver.
I am currently looking for a JDBC driver as a plugin for ingesting data from Cassandra to Logsash(Elasticsearch Stack, [https://www.elastic.co/guide/en/logstash/current/plugins-inputs-jdbc.html#_usage](url)  ) 
I wonder if this wrapper could work as a standalone JDBC driver for Cassandra?  Does the "Datastax Java Driver" has to be installed with the wrapper to make it work?

**Comment by maximevw** [comment]
Hello @fiddleboy,
Sorry for the delay... I missed your question.

It should be possible to use it as a standalone JDBC driver (but not tested as is for now, we only tested its usage as a dependency of a Java application) without additional requirements (the Datastax Java driver is embedded into the wrapper). A bundled version is provided (see [here](https://github.com/ing-bank/cassandra-jdbc-wrapper/releases/download/v4.5.0/cassandra-jdbc-wrapper-4.5.0-bundle.jar) for the latest one).

Based on the ELK documentation, you probably should configure something like that:
```
input {
  jdbc {
    jdbc_driver_library => "cassandra-jdbc-wrapper-4.5.0-bundle.jar"
    jdbc_driver_class => "com.ing.data.cassandra.jdbc.CassandraDriver"
    jdbc_connection_string => "jdbc:cassandra://host1--host2--host3:9042/keyspace?localdatacenter=DC1"
    ...
  }
}
```

Be careful, all the JDBC API is not implemented because Cassandra is a NoSQL database. So, some features could not work.

Do not hesitate to give your feedback on a such usage.
